lambda/update/correction.go

```diff
@@ -4,22 +4,38 @@ import (
 	"github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
 	"github.com/ministryofjustice/opg-data-lpa-store/internal/validate"
 	"github.com/ministryofjustice/opg-data-lpa-store/lambda/update/parse"
+	"strconv"
 	"time"
 )
 
 type Correction struct {
-	DonorFirstNames string
-	DonorLastName   string
-	DonorOtherNames string
-	DonorDob        shared.Date
-	DonorAddress    shared.Address
-	DonorEmail      string
-	LPASignedAt     time.Time
+	DonorFirstNames    string
+	DonorLastName      string
+	DonorOtherNames    string
+	DonorDob           shared.Date
+	DonorAddress       shared.Address
+	DonorEmail         string
+	LPASignedAt        time.Time
+	Index              *int
+	AttorneyFirstNames string
+	AttorneyLastName   string
+	AttorneyDob        shared.Date
+	AttorneyAddress    shared.Address
+	AttorneyEmail      string
+	AttorneyMobile     string
+	AttorneySignedAt   time.Time
 }
```

Comment:
maybe this would be neater, if it is going to grow, as

```go
type Correction struct {
  Donor DonorCorrection
  LPASignedAt time.time
  Attorney AttorneyCorrection 
  // or if you can only have one of these pointers, then you can nil check to see which type it is?
}

type DonorCorrection struct { /* ... */ }
type AttorneyCorrection struct { /* ... */ }
```

Reply:
then you'd also have the option of adding `(c DonorCorrection/AttorneyCorrection) Apply(lpa *shared.Lpa) []shared.FieldError` methods to break that up